src/screens/SettingsScreen.tsx

```diff
@@ -48,26 +48,40 @@ const SettingsScreen = ({ navigation }: ScreenProps) => {
   const dispatch = useAppDispatch()
   const theme = useTheme()
   const hasAvailableBiometrics = useBiometrics()
-
   const [
     { discreetMode, requireAuth, theme: currentTheme, currency: currentCurrency },
     currentNetworkName,
-    { metadataId: currentWalletId, authType }
+    activeWallet
   ] = useAppSelector((s) => [s.settings, s.network.name, s.activeWallet])
-  const biometricsEnabled = authType === 'biometrics'
 
-  const toggleBiometrics = async () => await dispatch(biometricsToggled({ enable: !biometricsEnabled }))
+  const isBiometricsEnabled = activeWallet.authType === 'biometrics'
+
+  const toggleBiometrics = async () => {
+    if (isBiometricsEnabled) {
+      await disableBiometrics(activeWallet.metadataId)
+      dispatch(biometricsDisabled())
```

Comment:
Yeah! This makes sense, and really simplify the state logic.

Reply:
Nice! Glad you see it the same way I do!